File: ackermann.swift
func ackermann(m: UInt, n: UInt) -> UInt {
    if m == 0 {
        return n + 1
    } else if m > 0 && n == 0 {
        return ackermann(m: m - 1, n: 1)
    } else {
        return ackermann(m: m - 1, n: ackermann(m: m, n: n - 1))
    }
}

print(ackermann(m: 4, n: 2))

File: ackermann.c
#include <stdio.h>
#include <stdlib.h>

unsigned long ackermann(unsigned long m, unsigned long n)
{
    if (m == 0) {
        return n + 1;
    } else if (m > 0 && n == 0) {
        return ackermann(m - 1, 1);
    } else if (m > 0 && n > 0) {
        return ackermann(m - 1, ackermann(m, n - 1));
    }
    exit(1);
}

int main()
{
    printf("%lu", ackermann(4, 2));
    return 0;
}
